include/nuttx/sensors/apds9922.h

```diff
@@ -0,0 +1,315 @@
+  struct adps9922_als_thresh
+                         thresh;      /* Upper and lower thresholds        */
+  enum apds9922_als_thresh_var
+                         thresh_var;  /* threshold variation               */
+  enum apds9922_als_int_mode
+                         int_mode;    /* Interrupt mode                    */
+  uint8_t                persistance; /* Num events before interrupt       */
+  uint32_t               als_factor;  /* Lux correction factor applied     */
+  uint32_t               range_lim;   /* % limit of ADC full range
+                                       * allowed in autogain mode.         */
+  bool                   autogain;    /* Auto gain mode on/off             */
+  enum apds9922_als_chan channel;     /* Visible or IR light channel       */
+};
+
+/* proximity sensor data setup data */
+
+struct apds9922_ps_setup_s
+{
+  enum apds9922_ps_rate   rate;        /* Measurement rate                  */
+  enum apds9922_ps_res    res;         /* Resolution, bits                  */
+  enum apds9922_ps_led_f  led_f;       /* LED modulation frequency          */
+  bool                    led_pk_on;   /* LED current peaking on/off        */
+  enum apds9922_ps_led_i  led_i;       /* LED pulsed current level          */
+  uint8_t                 pulses;      /* Number of LED pulses, 0-32        */
+  struct adps9922_ps_thresh
+                          thresh;      /* Upper and lower thresholds        */
+  uint16_t                cancel_lev;  /* Intelligent cancellation lev.     */
+  enum apds9922_ps_int_mode
+                          int_mode;    /* Interrupt mode                    */
+  uint8_t                 persistance; /* Num events before interrupt       */
+  enum apds9922_ps_notify notify;      /* States that cause a notify        */
+};
+
+/* data that can be read from proximity sensor */
+
+struct apds9922_ps_data
+{
+  uint16_t            ps;             /* Current prximity measure        */
+  bool                close;          /* Object is far (false) or close  */
+};
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+#ifdef __cplusplus
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+/****************************************************************************
+ * Name: apds9922_register
+ *
+ * Description:
+ *   Register the APDS9922 character devices.
+ *
+ * Input Parameters:
+ *   devpath_als - The full path to the driver to register for the als,
+ *   e.g., "/dev/als0"
+ *
+ *   devpath_ps - The full path to the driver to register for the als,
+ *   e.g., "/dev/ps0"
+ *
+ *   config - Pointer to the device configuration
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+struct i2c_master_s;
```

Comment:
Why forward declaration is needed here?

Reply:
Guess I based it on existing code that had it.